certbot/compat/filesystem.py

```diff
@@ -36,6 +43,67 @@ def chmod(file_path, mode):
         _apply_win_mode(file_path, mode)
 
 
+# Funtion os.makedirs will temporarily monkeypatch os.mkdir to replace it by filesystem.mkdir.
+# However, filesystem.mkdir also invokes os.mkdir. To avoid a looped reference, we take now a
+# reference of the original os.mkdir, to use it in filesystem.mkdir.
+_os_mkdir = os.mkdir
+
+
+def makedirs(file_path, mode=0o777, exists_ok=False):
+    # type: (str, int, bool) -> None
+    """
+    Rewrite of original os.makedirs function, that will ensure on Windows that given mode
+    is correctly applied.
+    :param str file_path: The file path to open
+    :param int mode: POSIX mode to apply on leaf directory when created, Python defaults
+                     will be applied if ``None``
+    :param bool exists_ok: If set to ``True``, do not raise exception if leaf directory
+                           already exists.
+    """
+    # As we know that os.mkdir is called internally by os.makedirs, we will swap the function in
+    # os module for the time of makedirs execution.
+    orig_mkdir_fn = os.mkdir
+    try:
+        os.mkdir = mkdir  # type: ignore
+        try:
+            os.makedirs(file_path, mode)
+        except (IOError, OSError) as err:
```

Comment:
Because `exists_ok` parameter does not exist on Python 2. Here is the usual way to get a consistent behavior on makedirs for both python versions. 

I followed here the API of Python 3, since it is the future. However you could argue that `exists_ok` is not used in Certbot codebase, so maybe you prefer that I just remove this feature?

Reply:
For both this and the discussion below, it's actually a good point that we should be hitting this code only on py3, since it's only running on windows. In which case I'm actually perfectly happy using py3-only code, especially since we're going to have to drop py2.7 soon anyway. So might as well pass `exists_ok` through.

Reply:
See general comment below.